README.md

```diff
@@ -1,2 +1,724 @@
+
+#### `Affjax`
+
+``` purescript
+type Affjax e a = Aff (ajax :: Ajax | e) (AffjaxResponse a)
+```
+
+The type for Affjax requests.
+
+#### `AffjaxOptions`
+
+``` purescript
+data AffjaxOptions :: *
+```
+
+Options type for Affjax requests.
+
+#### `AffjaxResponse`
+
+``` purescript
+type AffjaxResponse a = { response :: a, headers :: [ResponseHeader], status :: StatusCode }
+```
+
+The type of records that will be received as an Affjax response.
+
+#### `url`
+
+``` purescript
+url :: Option AffjaxOptions String
+```
+
+Sets the URL for a request.
+
+#### `method`
+
+``` purescript
+method :: Option AffjaxOptions Method
+```
+
+Sets the HTTP method for a request.
+
+#### `content`
+
+``` purescript
+content :: Option AffjaxOptions RequestContent
+```
+
+Sets the content to send in a request.
+
+#### `headers`
+
+``` purescript
+headers :: Option AffjaxOptions [RequestHeader]
+```
+
+Sets the headers to send with a request.
+
+#### `username`
+
+``` purescript
+username :: Option AffjaxOptions String
+```
+
+Sets the HTTP auth username to send with a request.
+
+#### `password`
+
+``` purescript
+password :: Option AffjaxOptions String
+```
+
+Sets the HTTP auth password to send with a request.
+
+#### `affjax`
+
+``` purescript
+affjax :: forall e a. Responsable a -> Options AffjaxOptions -> Affjax e a
+```
+
+Runs a request.
+
+#### `affjax'`
+
+``` purescript
+affjax' :: forall e a. Responsable a -> Options AffjaxOptions -> (Error -> Eff (ajax :: Ajax | e) Unit) -> (AffjaxResponse a -> Eff (ajax :: Ajax | e) Unit) -> Eff (ajax :: Ajax | e) Unit
+```
+
+Runs a request directly in Eff.
+
+#### `get`
+
+``` purescript
+get :: forall e a. Responsable a -> String -> Affjax e a
+```
+
+
+#### `post`
+
+``` purescript
+post :: forall e a. Responsable a -> String -> RequestContent -> Affjax e a
+```
+
+
+#### `post_`
+
+``` purescript
+post_ :: forall e. String -> RequestContent -> Affjax e Unit
+```
+
+
+#### `put`
+
+``` purescript
+put :: forall e a. Responsable a -> String -> RequestContent -> Affjax e a
+```
+
+
+#### `put_`
+
+``` purescript
+put_ :: forall e. String -> RequestContent -> Affjax e Unit
+```
+
+
+#### `delete`
+
+``` purescript
+delete :: forall e a. Responsable a -> String -> Affjax e a
+```
+
+
+#### `delete_`
+
+``` purescript
+delete_ :: forall e. String -> Affjax e Unit
+```
+
+
+
+## Module Network.HTTP.Method
+
+#### `Method`
+
+``` purescript
+data Method
+  = DELETE 
+  | GET 
+  | HEAD 
+  | OPTIONS 
+  | PATCH 
+  | POST 
+  | PUT 
+  | CustomMethod String
+```
+
+
+#### `eqMethod`
+
+``` purescript
+instance eqMethod :: Eq Method
+```
+
+
+#### `showMethod`
+
+``` purescript
+instance showMethod :: Show Method
+```
+
+
+#### `isOptionMethod`
+
+``` purescript
+instance isOptionMethod :: IsOption Method
+```
+
+
+#### `methodToString`
+
+``` purescript
+methodToString :: Method -> String
+```
+
+
+
+## Module Network.HTTP.MimeType
+
+#### `MimeType`
+
+``` purescript
+newtype MimeType
```

Comment:
Something like [purescript-mime](https://github.com/Fresheyeball/purescript-MIME) would be nice to use here eventually.